File: Typescript/titleToNumber.ts
function titleToNumber(columnTitle: string): number {
    let result = 0;
    let idx = 0;

    for (let i = columnTitle.length - 1; i >= 0; --i) {
        const d = columnTitle.charCodeAt(i) - 'A'.charCodeAt(0) + 1;
        result += Math.pow(26, idx) * d;
        ++idx;
    }

    return result;
}

File: C++/titleToNumber.cpp
class Solution 
{
public:
    int titleToNumber(string columnTitle) 
    {
        int result = 0;
        int idx = 0;
        for (int i = columnTitle.size() - 1; i >= 0; --i)
        {
            int d = columnTitle[i] - 'A' + 1;
            result += std::pow(26, idx) * d;
            ++idx;
        }
        return result;
    }
};